napari/utils/translations.py

```diff
@@ -0,0 +1,456 @@
+    Returns
+    -------
+    str
+        Localized `locale` and capitalized language name using `display_locale` as language.
+    """
+    # This is a dependency of the language packs to keep out of core
+    import babel
+
+    locale = locale if _is_valid_locale(locale) else DEFAULT_LOCALE
+    display_locale = (
+        display_locale if _is_valid_locale(display_locale) else DEFAULT_LOCALE
+    )
+    loc = babel.Locale.parse(locale)
+    return loc.get_display_name(display_locale).capitalize()
+
+
+def _is_valid_locale(locale: str) -> bool:
+    """
+    Check if a `locale` value is valid.
+
+    Parameters
+    ----------
+    locale: str
+        Language locale code.
+
+    Notes
+    -----
+    A valid locale is in the form language (See ISO-639 standard) and an
+    optional territory (See ISO-3166 standard).
+
+    Examples of valid locales:
+    - English: DEFAULT_LOCALE
+    - Australian English: "en_AU"
+    - Portuguese: "pt"
+    - Brazilian Portuguese: "pt_BR"
+
+    Examples of invalid locales:
+    - Australian Spanish: "es_AU"
+    - Brazilian German: "de_BR"
+    """
+    # This is a dependency of the language packs to keep out of core
+    import babel
+
+    valid = False
+    try:
+        babel.Locale.parse(locale)
+        valid = True
+    except babel.core.UnknownLocaleError:
+        pass
+    except ValueError:
+        pass
+
+    return valid
+
+
+def _run_process_and_parse(cmd: list):
+    """
+    Run a list of commands and return the result parsed form stdout.
+
+    Parameters
+    ----------
+    cmd : list
+        List of commands
+
+    Returns
+    -------
+    tuple
+        A tuple in the form ``(result_dict, message)``.
+    """
+    result = {"data": {}, "message": ""}
+    try:
+        p = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        stdout, stderr = p.communicate()
+        result = json.loads(stdout.decode('utf-8'))
+    except Exception:
+        result["message"] = (
+            traceback.format_exc() + "\n" + repr(stderr.decode('utf-8'))
+        )
+
+    return result["data"], result["message"]
+
+
+def get_installed_packages_locales() -> dict:
+    """
+    Get all napari extensions installed that contain locale data.
+
+    Returns
+    -------
+    dict
+        Ordered list of available language packs.
+        >>> {"package-name": locale_data, ...}
+
+    Examples
+    --------
+    - `entry_points={"napari.locale": "package-name = package_module"}`
+    - `entry_points={"napari.locale": "napari-git = napari_git"}`
+    """
+    cmd = [sys.executable, __file__, "_get_installed_language_pack_locales"]
+    found_package_locales, message = _run_process_and_parse(cmd)
+    return found_package_locales, message
+
+
+# --- Translators
+# ----------------------------------------------------------------------------
+class TranslationBundle:
+    """
+    Translation bundle providing gettext translation functionality.
+    """
```

Comment:
Can you add a doc string here for the `domain` and `locale` Parameters in the init

Reply:
Will do